### PR TITLE
fix(Checkbox): Changing visibility hover background color & unchecked fill color.

### PR DESCRIPTION
**Pull request by FlyersPh9**
- Changes visibility checkbox hover background color to allow background color to come through, matches borderless button's background fill color.  [Solves this issue in tree](https://github.com/iTwin/iTwinUI/pull/388#discussion_r754852780).
- Changes the fill color of the unchecked visibility checkbox glyph to better align with the standards.  [Discovered this issue in tree](https://github.com/iTwin/iTwinUI/pull/388#issuecomment-975690366).

iTwinUI_checkbox_Type_specialty_0_demo-special_0_desktopiTwinUI_dark_checkbox_Type_specialty_0_demo-special_0_desktop

### CODE REVIEW

File: src/inputs/checkbox.scss [hunk 1]
```diff
@@ -31,7 +31,7 @@
     // Adds hover to everything that is not disabled
     &:hover > input:enabled ~ .iui-checkbox-checkmark {
       @include themed {
```
Comment (bentleyvk) [inline src/inputs/checkbox.scss:33]: So for disabled checkmark, background is still there?

Comment (FlyersPh9) [reply src/inputs/checkbox.scss:33]: This is only for hover on enabled visibility checkboxes.  All disabled checkboxes have the same background color (hover or no hover) across all checkboxes.

File: src/inputs/checkbox.scss [hunk 2]
```diff
@@ -58,5 +58,11 @@
       opacity: 1;
       transition: opacity $iui-speed-fast ease;
     }
+
+    input:enabled ~ .iui-checkbox-checkmark > svg > .iui-uncheck {
```
Comment (mayank99) [inline src/inputs/checkbox.scss:62]: You don't need `:enabled` here, as disabled checkboxes already have a high specificity which should override this if you remove `> svg`.

image

But I did have to reduce specificity in one other place in `checkbox-radio`.
image

File: src/inputs/checkbox.scss (hunk 1 above)
Comment (mayank99) [reply src/inputs/checkbox.scss:33]: Then it won't really solve this issue.
image

Unless that is the intended look

Comment (FlyersPh9) [reply src/inputs/checkbox.scss:33]: I believe Vilius' screenshot is when hovered.  Here is a screenshot of this PR's checkboxes with a pink background.  Top checkbox is hovered (allows some of the pink bg to come thru due to opacity) and bottom checkbox has no hover.
Screen Shot 2021-11-23 at 10 22 37 AM

File: src/inputs/checkbox.scss (hunk 2 above)
Comment (FlyersPh9) [reply src/inputs/checkbox.scss]: Great suggestion, change is made!

**Review by mayank99 — APPROVED** [review]

**Review by bentleyvk — APPROVED** [review]

**Pull request merged**